### DISCUSSION TITLE
Where to find good tutorials? Books?

### DISCUSSION THREAD

**Discussion by sken78** [serenity-is/Serenity#6172] [Tricks and Samples]
This is a great framework but support is very poor. The learning curve is very hard for new developers. Many questions have no answers. I cannot find good tutorials. It should be a RAD but one spent weeks for trying to find a way how to add a simple Save Button to a GridEditorBase.

**Comment by VictorTomaili** [comment]
Just check the documentation 
https://serenity.is/docs

**Reply by sken78** [reply]
Thank you. But it is very poor documentation. In order to be able to use the full power of serenity you need to understand it. And to fight for this for weeks does not worth it. A powerfull framework with very limited possibilities to lern it.

**Reply by wezmag** [reply]
The source code itself is also a very good resource for learning.
